doc/secciones/05_analisis.tex

```diff
@@ -11,8 +11,178 @@ \chapter{Análisis del problema}
+correctamente modularizado. 
+
+El diseño dirigido por dominio (\textit{Domain Driven Design} en inglés,
+\textit{DDD} para acortar) es una de las metodologías más populares y efectivas
+para modelar un problema en el ámbito de la ingeniería del software. Mediante
+esta metodología se definen una serie de términos y patrones para modelar un
+problema como una abstracción de los componentes necesarios que puedan manejar
+el dominio del problema; en este caso, la extracción de información de TvTropes.
+Este modelado del problema nos permite definir las estructuras de datos,
+interfaces y los posibles errores o excepciones de forma general en el código
+que describan todas las partes importantes del dominio del problema y que luego
+se implementarán en el capítulo 6. De esta manera, se definirá un lenguaje
+unificado que todo el mundo implicado con este trabajo, tanto usuarios como
+clientes como desarrolladores, pueda entender y nos permita referirnos a todos
+los conceptos del dominio por términos fácilmente identificables y traducibles a
+código.
+
+\textit{DDD} define una serie de componentes que permiten dividir el dominio del
+problema en contextos muy definidos para facilitar que se creen unidades con
+responsabilidad única y que se puedan desarrollar de manera independiente. Los
+componentes que utilizaremos en este proyecto para modelarlo son: las entidades,
+los objetos valor, los agregados, los repositorios y los servicios. Las
+entidades son ítems o estructuras mutables y únicamente identificables que
+mantienen su identidad a lo largo del ciclo de vida, pero que pueden cambiar de
+estado. Por otro lado, los objetos valor son ítems inmutables no identificables,
+muy cercanos a tipos de datos que pueden ser más simples o complejos que, no
+obstante, describen las características de algo como una instancia; producir uno
+implica instanciar uno nuevo y no modificar el ya existente.
+
+Estos dos conceptos se unen mediante los agregados. Un agregado permite combinar
+y encapsular estas entidades y objetos valor, definiendo una interfaz común para
+ellos. Están principalmente definidos por una entidad raíz, cuyo identificador
+permite identificar al agregado. La lógica de negocio debe estar contenida en
+los agregados, que trabajarán con las entidades y objetos valor que la componen.
+Se puede entender un agregado como un objeto completo que realiza las
+funcionalidades que queremos y en el que sus datos no son accesibles desde
+fuera, solamente utilizables desde funciones. Es responsabilidad de las
+entidades y objetos valor el representar correctamente su información para que
+el agregado los pueda utilizar correctamente.
+
+En esta metodología entra también el concepto de repositorio, que almacena o
+persiste la información de un agregado, puesto que estos son la unidad de mayor
+importancia en \textit{DDD}. El repositorio añade una capa de abstracción para
+el almacenamiento de datos, generalmente usando el patrón repositorio, que
```

Comment:
referencia. ¿Es un patrón del "gang of five"?

Reply:
Añado referencia, pero segun estoy viendo es un patron posterior que no se define en el libro de gang of four